### DISCUSSION TITLE
11.0.0

### DISCUSSION THREAD

**Discussion by danwalmsley** [AvaloniaUI/Avalonia#12039] [General]
Our docs have had an overhaul also:
https://docs.avaloniaui.net/

See this guide on how to update your application.
https://docs.avaloniaui.net/docs/next/stay-up-to-date/upgrade-from-0.10

https://www.nuget.org/packages/Avalonia/11.0.0

New Features

A11y (Accessibility)
This version of Avalonia takes major strides in making applications more accessible. We've added support for various accessibility tools, making it easier for everyone to use Avalonia applications.

IME (Input Method Editor) Support
We have introduced support for Input Method Editor (IME), this allows onscreen keyboards and input in all languages.

Compositing Renderer
The new compositing renderer enhances the graphical rendering capabilities, providing a more robust, efficient, and flexible rendering pipeline.

WebAssembly (WASM) Support
WASM support allows Avalonia applications to run directly in the browser, broadening our platform support and giving developers more freedom and flexibility.

iOS and Android Support
We've added support for both iOS and Android platforms. Developers can now build and run Avalonia applications on two of the most popular mobile platforms.

Full Rich Text support
You can now fully render rich text allowing you to render complex documents.

With the Text Inlines feature, you can easily construct complex formatted text blocks. It enables you to add annotations, hyperlinks, and other inline elements in your text.

Smooth Virtualization (Reworked ItemsControl)
We've completely reworked the ItemsControl to provide smooth virtualization. This improves the efficiency and user experience for controls with many items.

Performance Improvements
Avalonia 11.0 comes with significant performance improvements, enhancing the speed and efficiency of your applications.

Control Themes, Nested Styles, and Theme Variants
We've added the ability to theme controls, apply nested styles, and choose theme variants. This makes it easier to build beautiful and consistent UIs.

Bitmap Effects
The Bitmap Effects feature lets you apply visual effects to bitmaps. You can add blur, drop shadows, and other effects to images.

3D Transforms
With 3D Transforms, you can now create stunning three-dimensional effects in your UI.

AOT (Ahead-Of-Time) Compilation and Trimming
AOT compilation enables your apps to run faster by compiling them before runtime. Trimming reduces the size of your apps by removing unused code.

GPU Interop
The GPU Interop feature allows Avalonia to work more efficiently with the GPU, improving rendering performance and visuals.

Experimental Metal Support
We're testing Metal support to improve performance on iOS and macOS devices.

<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.0.0'>11.0.0</a>.</em>

**Comment by Sciolo** [comment]
Sincere thanks to the Avalonia developers for all the effort they have put in. I just tested the bulk of functionality for desktop application and it's running perfectly. I'll be replacing the WPF version in a few days.

Kind Regards,
James

![aquila-avalonia-11](https://github.com/AvaloniaUI/Avalonia/assets/53260221/6a920c4f-28ff-4c67-86bc-ea2e6bd9f62b)

**Comment by ghost** [comment]
What a mega release, thanks to all contributors!

**Comment by francisferreira** [comment]
Does this version support UI Automation? We've been using version 0.10 so far, but that version didn't offer support to inspect and automate controls/elements. We were told that was available for 11.0-preview1 and newer, but at the time, due to our own project constraints, we were not allowed to build using a pre-release version of dependencies.

We have been looking forward to 11 since then. Hopefully that feature made it to this GA version!

**Reply by maxkatz6** [reply]
Yes, it does support accessibility APIs on Windows and macOS.

It also can be used with automation testing. But I would not recommend doing it, as it's not reliable for UI testing for multiple target platforms. See headless testing as an alternative: https://docs.avaloniaui.net/docs/next/concepts/headless/

**Comment by rings2006** [comment]
do the accessibility apis have to manually be implimented? if so where is the documentation on doing that? trying to help out the ryujinx team

**Comment by hhyyrylainen** [comment]
How do I use the new "Full Rich Text support"? I've been trying to search terms related to this but the only matches in the documentation I get are from the release notes itself. I did find some info on TextBlock Inlines, but those also don't seem to have any documentation whatsoever. Ever since seeing this new release announcement I had been looking forward to reimplementing one feature in my app with this new rich text support, but now I've hit a roadblock with not finding any other info than it being supported now.

**Reply by thevortexcloud** [reply]
I am not sure what that is a reference to, but I am guessing it has to do with this stuff: https://github.com/AvaloniaUI/Avalonia/tree/c3911177956897d7a5176f9d59932b308aebc0c9/src/Avalonia.Controls/Documents

In the case of a `Run` I am guessing you can put it inside a TextBlock if it works the same as it does in WPF. 

EDIT: This PR has some examples
https://github.com/AvaloniaUI/Avalonia/pull/6953

**Reply by hhyyrylainen** [reply]
That could be it and would make sense in the context as I saw references to documents where this feature was mentioned. Still I can't find any documentation about the documents feature either... The best I could find was this link to stackoverflow about wpf (about manually creating text parts and putting them in a text block): https://stackoverflow.com/questions/1959856/data-binding-the-textblock-inlines 

I see that example has just some basic formatting things. I guess the feature I'd most want to use (hyperlinks) is still missing as there's nothing about that in the documents folder, and there's this open discussion: https://github.com/AvaloniaUI/Avalonia/discussions/8818 linking to this issue: https://github.com/AvaloniaUI/Avalonia/issues/10186

**Reply by thevortexcloud** [reply]
Avalonia is missing some documentation for things. But a lot of it works exactly the same as WPF (some of it is a near line by line port from WPF/UWP/whatever). So I end up reading WPF's documentation sometimes.

But yeah, looks like you can't do inline hyperlinks yet. You could make a custom hyperlink textblock fairly easily though. You just need to respond to it being clicked on. FluentAvalonia (which I have never used) also apparently has a Hyperlink button in it.

https://github.com/amwx/FluentAvalonia

The implementation is very simple: https://github.com/amwx/FluentAvalonia/blob/30b4379a9c6a7fb2dde62442932e6bc985a23058/src/FluentAvalonia/UI/Controls/Button/HyperlinkButton.cs#L40

But it's also not particularly cross platform either. So it may not work for you.

**Reply by hhyyrylainen** [reply]
I've made a custom solution where I put text and buttons (with styling as stripped as I managed) into a flow container. It kind of works but sometimes the wrapping and element heights get pretty crazy, which is why I've been waiting for a proper rich text solution that would just easily behave nicely.

**Reply by maxkatz6** [reply]
I will leave it here: https://github.com/AvaloniaUI/avalonia-docs/issues/324

Note: we **do not** support editable inlines. Although, something like [AvaloniaEdit](https://github.com/AvaloniaUI/AvaloniaEdit) might be helpful in this case.
